sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -579,41 +690,83 @@ public Read withTemplateCompatibility() {
         BATCH
     }
 
+    /**
+     * An enumeration type for the method to be used when reading data from BigQuery.
+     */
+    public enum Method {
```

Comment:
Would recommend to mark all the methods associated with the new method `@Experimental` so we can change them later.

Reply:
Done.